### PR TITLE
Adding support for JSON and RAW JSON

### PR DESCRIPTION
**Pull request by yonatangross**
Hi Luke,

I hope you're doing well. It's been a while since we last connected.

I'm currently on reserve duty in the army, and unfortunately, my access to reliable Wi-Fi is quite limited. This has prevented me from building the code.

Nonetheless, I've managed to push through some updates. I've added support for both JSON and Raw JSON inputs and the corresponding tests and updated the README to reflect these changes.

I would appreciate your feedback on these updates. I'm looking forward to refining the implementation as soon as I have the means to build the project.

Best,

Yonatan.

### CODE REVIEW

**Comment by lukemartin** [comment]
Hey @yonatangross 👋 

Thanks for the contribution! Before I get a chance to look at the code, I'm just wondering what the motivation is for supporting another format? Was cURL not working out for you?

Thanks!
Luke

**Comment by yonatangross** [comment]
Hey @lukemartin, 

Some of our developers are using often these formats as well and that could save some time and make it more commonly use.

File: src/lib/utils.ts [hunk 1]
```diff
@@ -2,41 +2,68 @@ import lzString from 'lz-string';
 const { decompressFromEncodedURIComponent, compressToEncodedURIComponent } = lzString;
 
 const parseQuery = (
-	input: string
```
Comment (lukemartin) [inline src/lib/utils.ts:5]: Looks like the formatting has been changed here, probably just due to prettier not running. Once you're able to build the project properly, it should format through your IDE. Or you can run `yarn format` to fix them in one go.

File: src/lib/utils.ts [hunk 2]
```diff
@@ -2,41 +2,68 @@ import lzString from 'lz-string';
...
+    input: string
 ): {
-	error: boolean;
-	query: string;
-	variables: string;
-	operationName: string;
-	queryString: string;
+    error: boolean;
+    message: string;
```
Comment (lukemartin) [inline src/lib/utils.ts:8]: It doesn't look like `message` is being used anywhere. Should it be added to the UI?

File: tests/example.spec.ts [hunk 3]
```diff
@@ -18,6 +18,79 @@ const SAMPLE_OUTPUT_QUERY = `query GetRepository($name: String!, $owner: String!
 
 const SAMPLE_OUTPUT_VARIABLES = `{\n  "name": "name",\n  "owner": "owner"\n}`;
 
+const JSON_TEST_EXPECTED_VARIABLES = `{
+	"orgId": 15,
+	"searchPattern": "232f1350ff01 "
+  }`;
+
+const JSON_TEST_EXPECTED_QUERY = `query GetSuggestions($orgId: ID!, $searchPattern: String!, $contentTypeIds: [ID!]) {
```
Comment (lukemartin) [inline tests/example.spec.ts:26]: I'd suggest making this a bit more generic so it doesn't reflect any real-world queries which you might not want to expose... :)

**Review by lukemartin — CHANGES_REQUESTED** [review]
I'll admit I still don't understand the use-case here. Is it so that a query can be pulled from logs (rather than a browser) and added?

In any case, I'm happy to add it in if it's useful. Added a few comments for you to check out ✌️

File: tests/example.spec.ts (hunk 3 above)
Comment (yonatangross) [reply tests/example.spec.ts:26]: Agreed, I'll change as soon as I'm home.

File: src/lib/utils.ts (hunk 2 above)
Comment (yonatangross) [reply src/lib/utils.ts:8]: Probably, I'll fix it at home.

**Review by yonatangross — COMMENTED** [review]
Modified test to use generic nonrelated data & formatting.

**Review by yonatangross — COMMENTED** [review]
fixed testing and formating.

**Comment by yonatangross** [comment]
@lukemartin can you please check? :)

**Review by lukemartin — APPROVED** [review]

**Comment by lukemartin** [comment]
@yonatangross Approved and merged. Thanks!

**Pull request merged**